test/OpenTelemetry.Contrib.Instrumentation.MassTransit.Tests/MassTransitInstrumentationTests.cs

```diff
@@ -261,9 +265,9 @@ public async Task MassTransitInstrumentationTestOptions()
                 }
             }
 
-            Assert.Equal(4, activityProcessor.Invocations.Count);
+            Assert.Equal(8, activityProcessor.Invocations.Count);
```

Comment:
By default, the instrumentation should create 4 activities per message (publish, receive, consume, and handle). In this test, we are filtering tracing operations and should create only 1 activity. In the OpenTelemetry 0.7 it executed exactly one invocation. But in the current version, it returns 4 StartActivity events and 1 EndActivity event. 
@cijothomas  is it correct behavior?

Reply:
This is the correct behavior @alexvaluyskiy. With the new change to `TracerProviderSdk` to support legacy activities, the `Processor.OnStart` will be called for any legacy activity when it starts if it satisfies these conditions:
- `activity.OperationName` is added as a Legacy Source to `TracerProviderSdk`
- `activity.IsAllDataRequested` is `true` (this is set by the sampler)

When you set `activity.IsAllDataRequested` to `false` in the `OnStartActivity` method of the listener after filtering out the OperationNames that are not contained in `TracedOperations`, it ensures that `Processor.OnEnd` which is also responsible for exporting is not called for such activities.

You already have an `Assert` which ensures that there are exactly two invocations on the Processor with the required Activity OperationName.  If you'd like, you could go a little more specific in your testing to assert that the `Processor.OnEnd`  is only called for the activity whose OperationName is contained in `TracedOperations`. So in this test you could add something like this:
```
Assert.Single(activityProcessor.Invocations, invo => invo.Method.Name == "OnEnd");
var onEndProcessedActivity = activityProcessor.Invocations.FirstOrDefault(invo => invo.Method.Name == "OnEnd").Arguments[0] as Activity;
Assert.Equal(operationName, onEndProcessedActivity.OperationName);
```

Reply:
Thank you for the explanation. I've changed the assertions